project-repo/tests.md

```diff
@@ -3,39 +3,39 @@
 ## General principles
 
 * Any new code/feature should be accompanied by tests.
-* Tests should be baked by coverage report done in CodeCov. Any uncovered branch should be understood and tried to be covered if needed. We should aim at the highest coverage as possible.
-* Prefer package/module level testing to unit tests. Try to test only with the external API of this scope (if the language allows you to do this and only poke few holes, try it) and only go on the unit test level when this is needed (a complex, local, function). This allows you to assess your API and is a good documentation for anyone who need to use this package.
-* We do not encourage TDD, which forces to crystallize the API way too early in the design process. We thus compensate the lack of guided code path testability assurance:
-  * by using coverage extensively to ensure that the tests is going through the desired pass.
-  * By reversing, the "want" condition, to ensure that the test is failing as expected. It’s also a good time to assess and check that the returned error and log messages make sense.
-* Do not test only the happy path, ensure at least one error case, if the method can return an error, is covered. Try to think how you can trigger more error cases with an inconsistent file system state for instance (file being read only when you expect to write to it…).
-* One easy way to expand and facilitate adding tests in the future is to use a conjunction of table testing and golden files.
-* Apart when bootstrapping a project, the tests should always pass in the main branch.
+* Tests should be baked by coverage report done in CodeCov. Any uncovered branch should be understood, and covered if needed. We should aim for achieving the highest coverage possible.
+* Prefer package/module level testing to unit tests. Try to test only with the external API of this scope (if the language allows you to do this and only poke few holes, try it), and only go on the unit test level when this is needed (a complex, local, function). This approach allows you to assess your API, and is a good documentation for anyone who needs to use this package.
+* We do not encourage TDD, which forces crystalising the API way too early in the design process. We thus compensate the lack of guided code path testability assurance by
```

Comment:
Using british english :england:, I see :) (unsure we are consistent in that btw, but I try too)

Reply:
Indeed, I tried :) I myself don't use British English regularly either (I use Canadian English, a hybrid/mix between British and American), but yeah since it's company policy I tried to make things a bit more consistent here. :)